File: sorting/bubblesort/components/bubblesort.rb
class Array
  def bubblesort
    len = self.length
    while true
      bool = true
      0.upto(len - 2) do |x|
        bool, self[x], self[x+1] = false, self[x+1], self[x] if self[x] > self[x+1]
      end
      break if bool
    end
    return self
  end
end

File: sorting/bubblesort/components/bubblesort.js
export default function BubbleSort(array){
  let len = array.length;
  while(true){
    let bool = true;
    for(let i = 0; i < len - 1; i++){
      let a = array[i], b = array[i+1];
      if(a > b) bool = false, array[i] = b, array[i+1] = a;
    }
    if(bool) break;
  }
  return array;
}
